edx_name_affirmation/views.py

```diff
@@ -112,14 +137,71 @@ def post(self, request):
                     proctored_exam_attempt_id=request.data.get('proctored_exam_attempt_id', None),
                     status=request.data.get('status', VerifiedNameStatus.PENDING)
                 )
-                response_status = status.HTTP_200_OK
+                response_status = http_status.HTTP_200_OK
                 data = {}
             except VerifiedNameMultipleAttemptIds as exc:
-                response_status = status.HTTP_400_BAD_REQUEST
+                response_status = http_status.HTTP_400_BAD_REQUEST
                 data = {"detail": str(exc)}
         else:
-            response_status = status.HTTP_400_BAD_REQUEST
+            response_status = http_status.HTTP_400_BAD_REQUEST
+            data = serializer.errors
+        return Response(status=response_status, data=data)
+
+    def patch(self, request):
+        """
+        Update verified name status
+        """
+        if not request.user.is_staff:
+            return Response(
+                status=http_status.HTTP_403_FORBIDDEN,
+                data={'detail': 'Must be a staff user to update verified name status.'}
+            )
+
+        serializer = UpdateVerifiedNameSerializer(data=request.data)
+
+        if serializer.is_valid():
+            username = request.data.get('username')
+            user = get_user_model().objects.get(username=username)
+            try:
+                update_verified_name_status(
+                    user,
+                    request.data.get('status'),
+                    request.data.get('verification_attempt_id', None),
+                    request.data.get('proctored_exam_attempt_id', None)
+                )
+                response_status = http_status.HTTP_200_OK
+                data = {}
+            except (VerifiedNameAttemptIdNotGiven, VerifiedNameMultipleAttemptIds) as exc:
+                response_status = http_status.HTTP_400_BAD_REQUEST
+                data = {'detail': str(exc)}
+            except VerifiedNameDoesNotExist as exc:
+                response_status = http_status.HTTP_404_NOT_FOUND
+                data = {'detail': str(exc)}
+        else:
+            response_status = http_status.HTTP_400_BAD_REQUEST
             data = serializer.errors
+
+        return Response(status=response_status, data=data)
+
+    def delete(self, request, verified_name_id):
+        """
+        Delete verified name
+        /edx_name_affirmation/v1/verified_name/{verified_name_id}
+        """
+        if not request.user.is_staff:
+            return Response(
+                status=http_status.HTTP_403_FORBIDDEN,
+                data={'detail': 'Must be a staff user to delete a verified name.'}
+            )
+
+        try:
+            delete_verified_name(verified_name_id)
+            response_status = http_status.HTTP_200_OK
```

Comment:
Since you are not sending the deleted item back to frontend, the HTTP Status code is 204 instead of 200. See https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/DELETE